src/Adapter/MSTest.TestAdapter/Extensions/ExceptionExtensions.cs

```diff
@@ -32,6 +32,10 @@ internal static Exception GetRealException(this Exception exception)
         return exception;
     }
 
+    internal static bool IsOperationCanceledExceptionFromToken(this Exception ex, CancellationToken cancellationToken)
+        => (ex is OperationCanceledException oce && oce.CancellationToken == cancellationToken)
+        || (ex is AggregateException aggregateEx && aggregateEx.InnerExceptions.OfType<OperationCanceledException>().Any(oce => oce.CancellationToken == cancellationToken));
```

Comment:
Could we have an AggregateException with another inner AggregateException?

In that case, would it be better to change this to:

```csharp
    internal static bool IsOperationCanceledExceptionFromToken(this Exception ex, CancellationToken cancellationToken)
        => (ex is OperationCanceledException oce && oce.CancellationToken == cancellationToken)
        || (ex is AggregateException aggregateEx && aggregateEx.InnerExceptions.Any(innerEx => innerEx.IsOperationCanceledExceptionFromToken(cancellationToken));
```

Reply:
I'm not aware this could happen. Maybe using Flatten would be best

Reply:
I have tried a few things and cannot find a single scenario where this happens so I'll keep current logic.